plugins/modules/external_account.py

```diff
@@ -0,0 +1,381 @@
+      password: pass123!
+
+- name: Update AWS Access key credentials
+  cloudera.cluster.external_account:
+    host: example.cloudera.com
+    port: "7180"
+    username: "jane_smith"
+    password: "S&peR4Ec*re"
+    name: access_key_1
+    state: present
+    type: AWS
+    category: AWS_ACCESS_KEY_AUTH
+    params:
+      aws_access_key: access_key2
+      aws_secret_key: secret_key2
+
+- name: Remove basic authentication credentials
+  cloudera.cluster.external_account:
+    host: example.cloudera.com
+    port: "7180"
+    username: "jane_smith"
+    password: "S&peR4Ec*re"
+    name: Jane
+    state: absent
+
+"""
+
+RETURN = r"""
+---
+external_account:
+    description: Details of the external account created, updated, or retrieved.
+    type: dict
+    elements: complex
+    returned: always
+    contains:
+        name:
+            description: Represents the initial name of the account.
+            type: str
+            returned: always
+        displayName:
+            description: A modifiable label to identify this account for user-visible purposes.
+            type: str
+            returned: always
+        createdTime:
+            description: The time of creation for this account.
+            type: str
+            returned: always
+        lastModifiedTime:
+            description: The last modification time for this account.
+            type: str
+            returned: always
+        typeName:
+            description: The Type ID of a supported external account type.
+            type: str
+            returned: always
+        accountConfigs:
+            description: The configuration options for this account.
+            type: list
+            elements: dict
+            returned: always
+"""
+
+
+class ClouderaExternalAccount(ClouderaManagerModule):
+    def __init__(self, module):
+        super(ClouderaExternalAccount, self).__init__(module)
+
+        # Set the parameters
+        self.name = self.get_param("name")
+        self.category = self.get_param("category")
+        self.type = self.get_param("type")
+        self.params = self.get_param("params")
+        self.state = self.get_param("state")
+
+        # Initialize the return values
+        self.external_account_output = []
+        self.changed = False
+        # Execute the logic
+        self.process()
+
+    @ClouderaManagerModule.handle_process
+    def process(self):
+        api_instance = ExternalAccountsResourceApi(self.api_client)
+        existing = []
+        self.params = {
+            key: value for key, value in self.params.items() if value is not None
+        }
+        try:
+            existing = api_instance.read_account(self.name).to_dict()
+        except ApiException as ex:
+            if ex.status == 400:
+                pass
+            else:
+                raise ex
+
+        if self.state == "present":
+            try:
+                if existing:
+                    if not self.module.check_mode:
```

Comment:
Also need to add `diff` mode for everything but the secret keys, passwords, etc.

See:
- https://github.com/cloudera-labs/cloudera.cluster/blob/devel/plugins/modules/cm_service.py#L603-L608
- https://github.com/cloudera-labs/cloudera.cluster/blob/devel/plugins/modules/cm_service.py#L659-L661